server/modules/surveyRdb/repository/dataView/read.js

```diff
@@ -46,29 +148,36 @@ export const runSelect = async (
 
   const select = `
     SELECT 
-        ${colParamNames.join(', ')}
+        ${selectFields.join(', ')}
     FROM 
-        $/schemaName:name/.$/tableName:name/
+        ${fromTables.join(' ')}
     WHERE 
-      ${DataTable.colNameRecordCycle} = $/cycle/
+      ${viewDataNodeDef.columnRecordCycle} = $/cycle/
       ${R.isNil(filterClause) ? '' : `AND ${filterClause}`}
     ORDER BY 
-        ${R.isEmpty(sortParams) ? '' : `${sortClause}, `}date_modified DESC NULLS LAST
+        ${R.isEmpty(sortParams) ? '' : `${sortClause}, `}${viewDataNodeDef.columnDateModified} DESC NULLS LAST
     ${R.isNil(limit) ? '' : 'LIMIT $/limit/'}
     OFFSET $/offset/`
 
-  const params = {
+  const queryParams = {
     cycle,
     ...filterParams,
-    ...colParams,
     ...sortParams,
-    schemaName,
-    tableName,
+    schema: viewDataNodeDef.schema,
+    tableName: viewDataNodeDef.name,
     limit,
     offset,
   }
 
-  return queryStream ? new dbUtils.QueryStream(dbUtils.formatQuery(select, params)) : client.any(select, params)
+  const result = queryStream
+    ? new dbUtils.QueryStream(dbUtils.formatQuery(select, queryParams))
+    : await client.map(select, queryParams, _dbTransformCallbackSelect(viewDataNodeDef, editMode, nodeDefCols))
+
+  return {
+    result,
+    // Consider only user selected fields
+    selectFields: nodeDefCols.map((nodeDefCol) => new ColumnNodeDef(viewDataNodeDef, nodeDefCol).names).flat(),
```

Comment:
do you really need to return the selectFields?

Reply:
uhm, no, we can calculate them in the manager